### PR TITLE
Chore/sof 6426

### PR DESCRIPTION
**Pull request by spectre007**
This PR concerns the implementation of a new model tree based on asset files.

### CODE REVIEW

File: assets/dft/gga.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+default: true
```
Comment (timurbazhirov) [inline assets/dft/gga.yml:1]: Nit, but maybe `isDefault` here and elsewhere?

File: build_model_tree.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+};
+
+const getAssetFiles = (currentPath, assetExtension = ".yml") => {
+    return fs
+        .readdirSync(currentPath)
+        .filter((dirItem) => path.extname(dirItem) === assetExtension);
+};
+
+/**
+ * Check whether asset file is an auxiliary file.
```
Comment (timurbazhirov) [inline build_model_tree.js:34]: What is Auxiliary file?

File: build_model_tree.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    };
+};
+
+/**
+ * Build tree from a directory containing assets.
+ * @param {string} rootPath - path to tree root directory
+ * @param {string} rootLabel - label of root node
+ * @returns {{parent: undefined, data: undefined, children: *, label: string}}
+ */
+const build_tree = (rootPath, rootLabel = "root") => {
```
Comment (timurbazhirov) [inline build_model_tree.js:112]: Why is this snake case?

File: model_tree.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    label: "",
+    children: [
+        {
+            parent: "",
+            label: "/dft",
+            data: { type: { slug: "dft", name: "Density Functional Theory", default: "dft" } },
+            children: [
+                {
+                    parent: "/dft",
+                    label: "/dft/gga",
```
Comment (timurbazhirov) [inline model_tree.js:11]: Should these be `parentPath` and `path` instead of `parent and `label`?

File: model_tree.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                        subtype: {
+                            slug: "gga",
+                            name: "Generalized Gradient Approximation",
+                            default: "gga",
+                        },
+                    },
+                    children: [
+                        [
+                            {
+                                label: "/dft/pbe",
```
Comment (timurbazhirov) [inline model_tree.js:23]: Should be `/dft/gga/pbe`

File: build_model_tree.js (hunk 2 above)
Comment (spectre007) [reply build_model_tree.js:34]: Per our discussion, I moved the auxiliary files to a dedicated directory at the top level: `manifest`

File: model_tree.js [hunk 6]
```diff
@@ -0,0 +1,250 @@
+module.exports = {
```
Comment (timurbazhirov) [inline model_tree.js:1]: Do we want this to be committed and pretty formatted or autogenerated in post-install?

File: build_model_tree.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        parentPath: undefined,
+        path: rootNodePath,
+        data: undefined,
+        children,
+    };
+};
+
+const modelTree = buildTree(ASSET_PATH, "");
+
+fs.writeFileSync("./model_tree.js", `module.exports = ${JSON.stringify(modelTree)}`, "utf8");
```
Comment (timurbazhirov) [inline build_model_tree.js:117]: I suggest we explain the logic behind how the asset folders and tree work together into the README

File: model_tree.js (hunk 6 above)
Comment (spectre007) [reply model_tree.js:1]: I think autogenerated post-install should would be a good solution.

File: README.md [hunk 8]
```diff
@@ -55,3 +55,14 @@ present in Subworkflow units in the `WoDe` package ecosystem. The entities provi
...
+The `MoDe` package contains asset files defining a _model tree_, whereby each file
+defines a parts of the resulting model data. During `postinstall`, these files are processed
+by the `build_model_tree.js` script and a single model tree is created.
+The file path to a given asset file also determines its position in the tree using
+the following rules:
+- The asset folder `assets` represents the root node.
+- Asset files are expected to have the `.yml` file extension.
+- Each asset file defines a node.
+- Each directory having the same basename as an asset file defines the children of the corresponding node.
+- Asset file names ending in `.nodes.yml` define multiple children at once.
```
Comment (timurbazhirov) [inline README.md:68]: 1. Should we add a reference to CateCom? The purpose of the tree is not clear from what's written at present.
2. Lines 49, 50 should be deleted and the following explanation 52-56 should be added to line 6 at the top

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**